crates/e2e/tests/e2e/limit_orders.rs

```diff
@@ -290,6 +302,563 @@ async fn single_limit_order_test(web3: Web3) {
+        .balance_of(trader_a.address())
+        .call()
+        .await
+        .expect("Couldn't fetch TokenB's balance");
+    assert_eq!(balance, U256::from(99_650_498_453_042_316_811_u128));
+
+    let balance = token_a
+        .balance_of(trader_b.address())
+        .call()
+        .await
+        .expect("Couldn't fetch TokenA's balance");
+    assert_eq!(balance, U256::from(50_175_363_672_226_073_522_u128));
+
+    // Drive orderbook in order to check the removal of settled order_b
+    maintenance.run_maintenance().await.unwrap();
+    solvable_orders_cache.update(0).await.unwrap();
+
+    let auction = create_orderbook_api().get_auction().await.unwrap();
+    assert!(auction.auction.orders.is_empty());
+
+    // Drive again to ensure we can continue solution finding
+    driver.single_run().await.unwrap();
+}
+
+async fn mixed_limit_and_market_orders_test(web3: Web3) {
+    shared::tracing::initialize_for_tests("warn,orderbook=debug,solver=debug,autopilot=debug");
+    shared::exit_process_on_panic::set_panic_hook();
+    let contracts = crate::deploy::deploy(&web3).await.expect("deploy");
+
+    let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+    let solver_account = Account::Local(accounts[0], None);
+    let trader_a = Account::Offline(PrivateKey::from_raw(TRADER_A_PK).unwrap(), None);
+    let trader_b = Account::Offline(PrivateKey::from_raw(TRADER_B_PK).unwrap(), None);
+
+    // Create tokens to trade
+    let token_a = deploy_mintable_token(&web3).await;
+    let token_b = deploy_mintable_token(&web3).await;
+
+    // Fund trader accounts
+    tx!(
+        solver_account,
+        token_a.mint(trader_a.address(), to_wei(1010))
+    );
+    tx!(
+        solver_account,
+        token_b.mint(trader_b.address(), to_wei(510))
+    );
+
+    // Create and fund Uniswap pool
+    tx!(
+        solver_account,
+        contracts
+            .uniswap_factory
+            .create_pair(token_a.address(), token_b.address())
+    );
+    tx!(
+        solver_account,
+        token_a.mint(solver_account.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_b.mint(solver_account.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_a.approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        token_b.approve(contracts.uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver_account,
+        contracts.uniswap_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(100_000),
+            to_wei(100_000),
+            0_u64.into(),
+            0_u64.into(),
+            solver_account.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Fund settlement contract
+    tx!(
+        solver_account,
+        token_a.mint(contracts.gp_settlement.address(), to_wei(1010))
+    );
```

Comment:
If we only want to avoid the rounding issue mentioned in the PR description we would only ever have to fund the settlement contract with a single wei per order with 0 fees. Using 1010 ETH seems a bit excessive. 😅
Note that we only still run into this issue because the single wei of `surplus_fee` hardcoded at the moment is so small that it still causes numerical issues (adjusting the `buy_token` price for such a small fee doesn't change the buy_token price at all leaving again no fee to cover the `NaiveSolver` rounding issue).  For this specific test it be enough to not fund the settlement contract and instead hard code a fee of 2 wei (big enough to change the adjusted price).
However, to be on safe side and never have to worry about this thing it's probably best to be a bit more generous with the hardcoded `surplus_fee` (maybe 1_000?). This value is already unrealistically small but at least it should be big enough to never cause numerical issues, making it unnecessary to ever fund the settlement contract for an e2e test.